src/coord/src/coord.rs

```diff
@@ -1200,6 +1200,25 @@ impl Coordinator {
             .filter(|(_, frontier)| frontier != &Antichain::new())
             .collect();
 
+        for (id, frontier) in since_updates.iter() {
+            let item = self.catalog.try_get_by_id(*id).map(|e| e.item());
+            if let Some(CatalogItem::Index(catalog::Index { on, .. })) = item {
+                if let CatalogItem::Table(catalog::Table {
+                    persist: Some(persist),
+                    ..
+                }) = self.catalog.get_by_id(on).item()
```

Comment:
it strikes me as a little odd that one of these uses try_get_by_id and other other get_by_id. I'd say either standardize on one of leave a comment on why they're different

Reply:
I think the current stuff is correct (the index might not exist, but if it does its table must).

Reply:
ill add the comment but thats the reasoning yup